framework/include/problems/FEProblemBase.h

```diff
@@ -1340,20 +1354,59 @@ class FEProblemBase : public SubProblem, public Restartable
    */
   const UserObject & getUserObjectBase(const std::string & name, const THREAD_ID tid = 0) const;
 
-  /**
-   * Get the Positions object by its name
-   * @param name The name of the Positions object being retrieved
-   * @return Const reference to the Positions object
-   */
-  const Positions & getPositionsObject(const std::string & name) const;
-
   /**
    * Check if there if a user object of given name
    * @param name The name of the user object being checked for
    * @return true if the user object exists, false otherwise
    */
   bool hasUserObject(const std::string & name) const;
 
+#ifdef MOOSE_KOKKOS_ENABLED
+  virtual void addKokkosUserObject(const std::string & user_object_name,
+                                   const std::string & name,
+                                   InputParameters & parameters);
+
+  /**
+   * Get the Kokkos user object by its name
+   * @param name The name of the Kokkos user object being retrieved
+   * @return Const reference to the Kokkos user object
+   */
+  template <class T>
+  T & getKokkosUserObject(const std::string & name) const
```

Comment:
I'm not the biggest fan that this is const and we're getting a non-const value. Do we need non-const values? We made this mistake a while ago in moose and we can't come back from it. Would be nice to not make it here.

Also - while I'm here - as I implement getters like this, I've been doing it like they do in libmesh.

You have a `queryXXX` method that gives you a pointer (that can be null).  Then you have a `getXXX` method that calls `queryXXX`, errors if it's null and otherwise dereferences. and then the `hasXXX` method just calls `queryXXX` and checks if it is null. Means that all of your actual checking will live in a single function, the `queryXXX` method.